File: backend/src/Accounting.Domain/Entities/Voucher.cs
using Accounting.Domain.Common;

namespace Accounting.Domain.Entities;

public class Voucher : BaseEntity
{
    public string VoucherNumber { get; set; } = string.Empty;
    public string VoucherType { get; set; } = "CP"; // Cash Payment
    public DateTime VoucherDate { get; set; } = DateTime.UtcNow;
    public string? Reference { get; set; }
    public string Narration { get; set; } = string.Empty;

    public decimal TotalDebit { get; set; }
    public decimal TotalCredit { get; set; }

    public bool IsPosted { get; set; } = false;
    public DateTime? PostedAtUtc { get; set; }

    // Navigation Properties
    public virtual ICollection<VoucherLine> Lines { get; set; } = new List<VoucherLine>();
    public virtual ICollection<LedgerTransaction> LedgerTransactions { get; set; } = new List<LedgerTransaction>();
}

File: app/src/main/java/com/example/domain/model/Voucher.kt
package com.example.domain.model

data class Voucher(
    val id: Long = 0,
    val voucherNumber: String,
    val voucherType: String = "CP",
    val voucherDate: Long,
    val reference: String = "",
    val narration: String = "",
    val isPosted: Boolean = false,
    val postedAt: Long? = null,
    val lines: List<VoucherLine> = emptyList(),
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val totalDebitCents: Long
        get() = lines.sumOf { it.debitCents }

    val totalCreditCents: Long
        get() = lines.sumOf { it.creditCents }

    val differenceCents: Long
        get() = totalDebitCents - totalCreditCents

    val isBalanced: Boolean
        get() = totalDebitCents > 0L && totalDebitCents == totalCreditCents
}
